main.go

```diff
@@ -85,47 +70,41 @@ func main() {
 	}
 
 	timer := time.NewTimer(0)
+	<-timer.C // Avoid to run command just after startup.
 	changes := startWatching(*watchPath)
-	lastRun := time.Time{}
-	lastChange := time.Now()
+	lastRun := time.Now()
+	lastChange := lastRun
 
 	for {
 		select {
 		case lastChange = <-changes:
-			timer.Reset(rebuildDelay)
-
-		case <-ui.rerun():
-			lastRun = run(ui)
-
-		case <-timer.C:
 			if lastRun.Before(lastChange) {
-				lastRun = run(ui)
+				timer.Reset(rebuildDelay)
 			}
+
+		case <-timer.C:
+			lastRun = run()
 		}
 	}
 }
 
-func run(ui ui) time.Time {
-	ui.redisplay(func(out io.Writer) {
-		cmd := exec.Command(flag.Arg(0), flag.Args()[1:]...)
-		cmd.Stdout = out
-		cmd.Stderr = out
-		if hasSetPGID {
-			var attr syscall.SysProcAttr
-			reflect.ValueOf(&attr).Elem().FieldByName(setpgidName).SetBool(true)
-			cmd.SysProcAttr = &attr
-		}
-		io.WriteString(out, strings.Join(flag.Args(), " ")+"\n")
-		start := time.Now()
-		if err := cmd.Start(); err != nil {
-			io.WriteString(out, "fatal: "+err.Error()+"\n")
-			return
-		}
-		if s := wait(start, cmd); s != 0 {
-			io.WriteString(out, "exit status "+strconv.Itoa(s)+"\n")
-		}
-		io.WriteString(out, time.Now().String()+"\n")
-	})
+func run() time.Time {
+	cmd := exec.Command(flag.Arg(0), flag.Args()[1:]...)
+	if hasSetPGID {
+		var attr syscall.SysProcAttr
+		reflect.ValueOf(&attr).Elem().FieldByName(setpgidName).SetBool(true)
+		cmd.SysProcAttr = &attr
+	}
+	fmt.Print(strings.Join(flag.Args(), " ") + "\n")
+	start := time.Now()
+	if err := cmd.Start(); err != nil {
```

Comment:
If I'm reading this right, the command's stderr and stdout will be dropped, which is not great for debugging.
The old code sent them both to stdout; passing them through to the parent's would also be ok; using `cmd.CombinedOutput` and only printing it in case of failure would be another possibility.

Reply:
Oops, it wasn't my intention to change the old behavior here. Let me check…

Reply:
Got it. I somehow assumed `cmd.Stdout` would be set to `os.Stdout` by default. Will fix in a second.